### ISSUE TITLE
appium flutter library but can't use commands with webview

### ISSUE THREAD

**Issue by jaylollipop** [appium/appium-flutter-driver#495]
my app is developed using flutter but some pages are webview ...
ex. run robot framework use appium flutter library can't click or input page webview

How do you have a way to do it? appiumflutterlibrary can be used for webview page ?

**Comment by denysord88** [comment]
Hi,
Did you try to switch context?
In Java we are using something like `driver.context("NATIVE_APP")` to use Appium elements location
`driver.context("WEBVIEW")` to interact with webview
`driver.context("FLUTTER")` to use FlutterFinder

**Comment by jaylollipop** [comment]
appium flutter library  no keyword "switch to context" 
How can I use switch to context based on main:appium flutter library?

> Hi, Did you try to switch context? In Java we are using something like `driver.context("NATIVE_APP")` to use Appium elements location `driver.context("WEBVIEW")` to interact with webview `driver.context("FLUTTER")` to use FlutterFinder

**Comment by KazuCocoa** [comment]
What client did you use?
You could find some client library's example to switch context: https://github.com/search?q=repo%3Aappium-userland%2Fappium-flutter-driver+NATIVE_APP+path%3A%2F%5Eexample%5C%2F%2F&type=code

**Comment by KazuCocoa** [comment]
https://appium.io/docs/en/latest/guides/context/

**Comment by TestMagic0989** [comment]
@KazuCocoa why this issue is closed and this link below just provide general information about context set and get
[](https://appium.io/docs/en/latest/guides/context/) not really solves the problem mentioned as title because when you use appium flutter driver it only gives you two type of context FLUTTER and NATIVE_APP in an arry after sending GET http request. So, do we have any real solution If In case when we automate Flutter app which is using component from NATIVE and some WEVIEW and It that case how to fix this webview switch context issue

**Comment by KazuCocoa** [comment]
WEBVIEW contexts depend on UIA2/XCUITest driver. This driver returns WEBVIEW contexts from them.

https://github.com/appium/appium-flutter-driver?tab=readme-ov-file#appium-flutter-driver-or-appium-uiautomator2xcuitest-driver

> WEBVIEW context manages the WebView contents over Appium UiAutomator2/XCUITest driver

I'm not sure how to configure in Flutter framework side (maybe Flutter team has their own documentation to make their WebView contents debuggable), but below links in each driver (links from https://github.com/appium/appium-flutter-driver?tab=readme-ov-file#commands-for-native_appwebview-context ) would show hints to test them out via Appium drivers.

https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#hybrid-mode
https://appium.github.io/appium-xcuitest-driver/latest/preparation/#webview-testing

**Comment by TestMagic0989** [comment]
@KazuCocoa I know already thats possible with [uiautomator2/ xcuitest-driver] and it allows you to switch to webview and its an option one can code in there framework by replacing the automationName as placeholder as an when they need to switch the driver from appium flutter driver to uiautomator2 or xcuitest-driver to inspect the webview.
One more possibility is to use UiSelector() to scroll in webview wiithout changing appium flutter driver however its not the best solution but it works for now given the situation we dont have any real solution as such when we use appium flutter driver

**Comment by KazuCocoa** [comment]
> when they need to switch the driver from appium flutter driver to uiautomator2 or xcuitest-driver to inspect the webview.

I meant this flutter driver got WEBVIEW contexts via UIA2/XCUITest driver internally. So if your app could get WEBVIEW contexts via uia2/xcuitest driver, the appium-flutter-driver also might be able to get them as well via uia2/xcuitest driver internally. Then, `automationName` can keep `flutter`.

https://github.com/appium/appium-flutter-driver/blob/7d57cc8363c7c321d36e9cc12e53fb000f53d18e/driver/lib/commands/context.ts#L34

If you meant get context request didn't go though uia2/xcuitest driver in flutter automationName, could you create a new issue with the full server log to check the behavior?

**Comment by TestMagic0989** [comment]
@KazuCocoaI I understand that when using flutter as the automationName in tests, and if we switch the context to NATIVE_APP, the system starts using UIAutomator2 for Android (or the native automation framework) to recognize elements. My understanding is that when this happens, we should be able to interact with WebView elements as well, by default, using native methods like UiScrollable and scrollIntoView() to inspect and locate WebView elements which I figured some now and it is working.

However, I also want to confirm that in some cases, explicit switching to the WebView context may still be necessary for more detailed or advanced interactions with specific Web elements inside the WebView.
So this being handled as well using flutter-appium-driver?

**Comment by KazuCocoa** [comment]
So, interacting with web elements in MATIVE_APP context? Then, it depends on Android OS side. Appium cannot inject such OS level recognition. Appium gets elements via AccessibilityNodeInfo by Android API. If web view elements are available via it (NATIVE_APP context) depends on OS side behavior.
So officially we provide WEBVIEW context to interact with WebView elements but no guarantees in NATIVE_APP context because the behavior is not injectable by Appium's layer.

**Comment by KazuCocoa** [comment]
https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#elements-cannot-be-found potentially helps, btw.
